src/Federation/Resolvers/Entities.php

```diff
@@ -33,27 +28,48 @@ public function __invoke($root, array $args, GraphQLContext $context, ResolveInf
     {
         $results = [];
 
-        /** @var array<string, array<int, array<string, mixed>>> $groupedRepresentations */
+        $representations = $args['representations'];
+        $representationHashes = array_map('serialize', $representations);
+
+        $assignResultsByHash = function ($result, string $hash) use ($representationHashes, &$results): void {
+            foreach ($representationHashes as $index => $h) {
+                if ($hash === $h) {
+                    $results[$index] = $result;
+                }
+            }
+        };
+
+        /**
+         * Firstly, representations are grouped by typename to allow assigning the correct resolver for each entity.
+         * Secondly, they are deduplicated based on their hash to avoid resolving the same entity twice.
```

Comment:
Should resolving be a concern of a specific entity resolver? For example the same representation can be resolved differently in the same request. A bit of an artificial example, but things like 
```
"representations": [
  {
      "__typename": RandomNumber
      "max": 100
  },
{
      "__typename": RandomNumber
      "max": 100
  }
]
```
will have to return 2 different random numbers.

Reply:
Since we are dealing with *entities* here, I would expect them to be consistent within a single request. For use cases like your example, dedicated custom queries seem more fitting.